### ISSUE TITLE
Version 1.0.0 does not install on RHEL 7 servers due to host config files not existing

### ISSUE THREAD

**Issue by piontekdd** [Dynatrace/Dynatrace-OneAgent-Puppet#24]
Trying to update from 0.5.0 to 1.0.0 to fix the Notify['Not reboot]' message coming out every single run. 1.0.0 fails on a fresh install because it isn't creating the directory where the new config files go. This seems like a dependency issue:

```
Notice: /Stage[main]/Profile::Dynatrace::Oneagent::Install/File[/opt/dynatrace]/owner: owner changed 'root' to 'rtc005'
Notice: /Stage[main]/Profile::Dynatrace::Oneagent::Install/File[/opt/dynatrace]/group: group changed 'root' to 'sys'
Notice: /Stage[main]/Profile::Dynatrace::Oneagent::Install/File[/opt/dynatrace/oneagent]/ensure: created
Notice: /Stage[main]/Profile::Dynatrace::Oneagent::Install/Firewall[850 dynatrace client]/ensure: created
Notice: /Stage[main]/Dynatraceoneagent::Install/Archive[/tmp/Dynatrace-OneAgent-Linux-latest.sh]/ensure: download archive from https://ncc510.dynatrace-managed.com/e/e5b75fad-801b-4df2-b36e-5c5a11c69394/api/v1/deployment/installer/agent/unix/default/latest/?Api-Token=ZJIEBMMaTjisLxaRsOLIT&arch=all to /tmp/Dynatrace-OneAgent-Linux-latest.sh  with cleanup
Error: Could not set 'present' on ensure: No such file or directory @ dir_s_mkdir - /var/lib/dynatrace/oneagent/agent/config/hostautotag.conf20200225-53695-17elc3y.lock (file: /etc/puppetlabs/code/environments/engineering/modules/forge/dynatraceoneagent/manifests/config.pp, line: 22)
Error: Could not set 'present' on ensure: No such file or directory @ dir_s_mkdir - /var/lib/dynatrace/oneagent/agent/config/hostautotag.conf20200225-53695-17elc3y.lock (file: /etc/puppetlabs/code/environments/engineering/modules/forge/dynatraceoneagent/manifests/config.pp, line: 22)
Wrapped exception:
No such file or directory @ dir_s_mkdir - /var/lib/dynatrace/oneagent/agent/config/hostautotag.conf20200225-53695-17elc3y.lock
Error: /Stage[main]/Dynatraceoneagent::Config/File[/var/lib/dynatrace/oneagent/agent/config/hostautotag.conf]/ensure: change from 'absent' to 'present' failed: Could not set 'present' on ensure: No such file or directory @ dir_s_mkdir - /var/lib/dynatrace/oneagent/agent/config/hostautotag.conf20200225-53695-17elc3y.lock (file: /etc/puppetlabs/code/environments/engineering/modules/forge/dynatraceoneagent/manifests/config.pp, line: 22)
Notice: /Stage[main]/Dynatraceoneagent::Service/File[/tmp/Dynatrace-OneAgent-Linux-latest.sh]: Dependency File[/var/lib/dynatrace/oneagent/agent/config/hostautotag.conf] has failures: true
```

manually creating /var/lib/dynatrace/oneagent/agent/config allows the install the work, but this should not be a requirement. I downgraded to 0.5.0 and it installs just fine.

**Comment by piontekdd** [comment]
Looks to me like this directory gets created by the oneagent installer:
```
readonly RUNTIME_BASE=/var/lib
readonly RUNTIME_DIR=${RUNTIME_BASE}/${BRANDING_PRODUCTNAME_LOWER}/${BRANDING_PRODUCTSHORTNAME_LOWER}
readonly PARTIAL_RUNTIME_DIR=${RUNTIME_BASE}/${BRANDING_PRODUCTNAME_LOWER}
readonly AGENT_RUNTIME_DIR=${RUNTIME_DIR}/agent
readonly AGENT_RUNTIME_CONF_PATH=${AGENT_RUNTIME_DIR}/config
```

This is called form service.pp, which is after config.pp is is called. I'd propose putting the Exec in install.pp where it likely belongs and leaving service.pp to only be for the service. This should solve the dependency problem.

**Comment by luisescobar89** [comment]
Thank you Bradd for reporting the issue and for proposing a solution, the issue is now fixed on version 1.1.0
